src/OSPSuite.R/Services/PopulationTask.cs

```diff
@@ -56,14 +71,42 @@ public DataTable PopulationTableFrom(IndividualValuesCache population, IModelCor
          return dataTable;
       }
 
-      private void addCovariates(IndividualValuesCache population, DataTable dataTable)
+      public IReadOnlyList<string> SplitPopulation(string populationFile, int numberOfCores, string outputFolder, string outputFileName)
       {
-         var individualIds = Enumerable.Range(0, population.Count).ToList();
+         var population = ImportPopulation(populationFile);
+         var populationData = PopulationTableFrom(population);
+         var dataSplitter = new PopulationDataSplitter(numberOfCores, populationData);
+         DirectoryHelper.CreateDirectory(outputFolder);
+         var outputFiles = new List<string>();
 
-         //add individual ids column
-         individualIds.Each(i => dataTable.Rows.Add(dataTable.NewRow()));
-         addColumnValues(dataTable, Constants.Population.INDIVIDUAL_ID_COLUMN, individualIds);
+         for (int i = 0; i < numberOfCores; i++)
+         {
+            var outputFile = Path.Combine(outputFolder, $"{outputFileName}_{i + 1}{Constants.Filter.CSV_EXTENSION}");
+            var rowIndices = dataSplitter.GetRowIndices(i).ToList();
+
+            //This is potentially empty if the number of individuals in the population is less than the number of cores provided
+            if(!rowIndices.Any())
```

Comment:
I hope R population runner will not crash if it will not find its input population file :)

Reply:
This is somewhat of a stupid example. Why woyld anyone run in // when the problem is small

Reply:
could happen even with bigger pops.
E.g. if you have 960 Individuals and 50 cores we will end up with 20 Individuals per cores and thus the last 2 cores will have nothing todo.

Reply:
hum yeah ok, Whatever then :)